File: conversions/typescript/Institution.ts
import { BaseElement } from './base_elements';

export interface Institution extends BaseElement {
  // Foundation
  premise?: string | null;
  foundDate?: number | null;
  endDate?: number | null;
  parentInstitutionId?: string | null;
  // Claim
  territoriesIds?: string[] | null;
  objectsIds?: string[] | null;
  creaturesIds?: string[] | null;
  legalIds?: string[] | null;
  // World
  situation?: string | null;
  cooperatesIds?: string[] | null;
  competitionIds?: string[] | null;
  constructsIds?: string[] | null;
  phenomenaIds?: string[] | null;
}

File: languages/csharp_unity/Institution.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Institution : BaseElement
{
    // Foundation
    public string Doctrine;
    public int? FoundingDate;
    public string ParentInstitutionId;
    // Claims
    public List<string> LegislationIds; = new List<string>();
    public List<string> ZonesIds; = new List<string>();
    public List<string> ObjectsIds; = new List<string>();
    public List<string> CreaturesIds; = new List<string>();
    // World
    public string Status;
    public List<string> AlliesIds; = new List<string>();
    public List<string> AdversariesIds; = new List<string>();
    public List<string> ConstructsIds; = new List<string>();
}
